mu4/domain/notation/internal/notationinteraction.h

```diff
@@ -118,6 +120,11 @@ class NotationInteraction : public INotationInteraction
     void setDropTarget(Element* el);
     bool dropCanvas(Element* e);
 
+    void applyDropPeletteElement(Ms::Score* score, Ms::Element* target, Ms::Element* e, Qt::KeyboardModifiers modifiers,
+                                 QPointF pt = QPointF(), bool pasteMode = false);
+    void cmdAddSlur(const Ms::Slur* slurTemplate = nullptr);
```

Comment:
do we really need them both?

Reply:
I didn’t understand the question, most of the elements are added through the drop operation, but for some elements their own adding procedures. 
At the moment, I tried to keep the algorithm as is, without changing it, so as not to mix the change in the structure of the code and the change in implementation.

Reply:
I've told about the addSlur and cmdAddSlur methods